src/node_jsvmapi.cc

```diff
@@ -465,22 +465,58 @@ namespace v8impl {
 
 }  // end of namespace v8impl
 
-// We define a macro here because, with full error handling, we may want to
-// perform additional actions at the top of our public APIs, such as bail,
-// returning an error code at the very top of each public API in the case that
-// there's already an exception pending in v8impl::TryCatch::lastException
+   // We define a macro here because, with full error handling, we may want to
+   // perform additional actions at the top of our public APIs, such as bail,
+   // returning an error code at the very top of each public API in the case that
+   // there's already an exception pending in v8impl::TryCatch::lastException
 #define NAPI_PREAMBLE(e) \
   v8impl::TryCatch tryCatch(v8impl::V8IsolateFromJsEnv((e)))
 
-napi_env napi_get_current_env() {
-  return v8impl::JsEnvFromV8Isolate(v8::Isolate::GetCurrent());
+   // Static last error returned from napi_get_last_error_info
+napi_extended_error_info static_last_error;
+
+// Warning: Keep in-sync with napi_errorvalue enum
+const char* error_messages[] = {
+  NULL,
+  "Invalid pointer passed as argument",
+  "An object was expected",
+  "A string was expected",
+  "Unknown failure"
+};
+
+const napi_extended_error_info* napi_get_last_error_info() {
+  // Wait until someone requests the last error information to fetch the error message string
+  static_last_error.error_message = error_messages[static_last_error.error_code];
```

Comment:
Should probably check that the error code is within range, and fallback to null or a default message if it's not. (Yeah as long as it's one of the enum values it *should* be in range, but it would be easy to add an enum value without adding a string, or pass in some other value to the function.)